jetstream/tests/jsm_test.ts

```diff
@@ -2164,3 +2166,43 @@ Deno.test("jsm - stream/consumer metadata", async () => {
 
   await cleanup(ns, nc);
 });
+
+Deno.test("jsm - consumer api action", async () => {
+  const { ns, nc } = await setup(jetstreamServerConf());
+  if (await notCompatible(ns, nc, "2.10.0")) {
+    return;
+  }
+
+  const jsm = await nc.jetstreamManager();
+  await jsm.streams.add({ name: "stream", subjects: ["a"] });
+
+  const config = {
+    ack_policy: AckPolicy.Explicit,
+    durable_name: "hello",
+  } as ConsumerConfig;
+
+  // testing the app, consumer update does an info
+  // so testing the underlying API
+  const api = jsm.consumers as ConsumerAPIImpl;
+  await assertRejects(
+    async () => {
+      await api.add("stream", config, ConsumerApiAction.Update);
```

Comment:
I guess this is for protection against breaking changes, but I'm not sure if `ConsumerApiAction` is clear enough.

If you do not want separate `update` `create` and `create_or_update` methods, maybe at least call the action `CreateOnly` `UpdateOnly` `CreateOrUpdate` so intention is clear?